sentry/src/main/java/io/sentry/adapters/TransactionContextsSerializerAdapter.java

```diff
@@ -0,0 +1,62 @@
+package io.sentry.adapters;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.SpanContext;
+import io.sentry.TransactionContexts;
+import io.sentry.protocol.App;
+import io.sentry.protocol.Browser;
+import io.sentry.protocol.Device;
+import io.sentry.protocol.Gpu;
+import io.sentry.protocol.OperatingSystem;
+import io.sentry.protocol.SentryRuntime;
+import java.lang.reflect.Type;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class TransactionContextsSerializerAdapter
+    implements JsonSerializer<TransactionContexts> {
+
+  private final @NotNull ILogger logger;
+
+  public TransactionContextsSerializerAdapter(@NotNull final ILogger logger) {
+    this.logger = logger;
+  }
+
+  @Override
+  public JsonElement serialize(
+      TransactionContexts src, Type typeOfSrc, JsonSerializationContext context) {
+    if (src == null) {
+      return null;
+    }
+
+    final JsonObject object = new JsonObject();
+    object.add(SpanContext.TYPE, context.serialize(src.getTrace(), Object.class));
```

Comment:
`ContextsDeserializerAdapter` uses `SpanContext.TYPE` and apparently after this PR, it'd not be necessary anymore, right?

Reply:
It will be, as SpanContext (aka "trace") is also a part of events.